APACHE_PROPOSAL.md

```diff
@@ -0,0 +1,326 @@
+
+#### Relationships with Other Apache Products:
+
+HugeGraph follows the Apache TinkerPop specification, uses Apache Commons, Apache HttpClient, and Apache HttpCore to implement the basic functions. Users can choose Apache Cassandra or Apache HBase as one of the storage backends of HugeGraph.
+
+#### A Excessive Fascination with the Apache Brand:
+
+Although we expect that the Apache brand may help attract more contributors, our interest in starting this project is based on the factors mentioned in the fundamentals section. We are interested in joining ASF to increase our connections in the open source world. Based on extensive collaboration, it is possible to build a community of developers and committers that live longer than the founder.
+
+### Documentation
+
+HugeGraph documentation is provided on https://hugegraph.github.io/hugegraph-doc/ in Simplified Chinese, the complete English version of the documentation is being prepared.
+
+### Initial Source
+
+This project consists of 2 core sub-projects and 8 complementary sub-projects, all of them are hosted by [GitHub  hugegraph organization](https://github.com/orgs/hugegraph/repositories) since 2018, the codes are already under Apache License Version 2.0. The git address of sub-project repositories are as follows:
+
+1. The graph database repository `hugegraph`, core sub-project, including graph server, graph engine and graph storage: https://github.com/hugegraph/hugegraph
+2. The graph computing repository `hugegraph-computer`, core sub-project, including graph computing and graph algorithm s: https://github.com/hugegraph/hugegraph-computer
+3. `hugegraph-common`: https://github.com/hugegraph/hugegraph-common
+4. `hugegraph-client`: https://github.com/hugegraph/hugegraph-client
+5. `hugegraph-doc`: https://github.com/hugegraph/hugegraph-doc
+6. `hugegraph-loader`: https://github.com/hugegraph/hugegraph-loader
+7. `hugegraph-tools`: https://github.com/hugegraph/hugegraph-tools
+8. `hugegraph-rpc`: https://github.com/hugegraph/hugegraph-rpc
+9. `hugegraph-hubble`: https://github.com/hugegraph/hugegraph-hubble
+10. `hugegraph-test`: https://github.com/hugegraph/hugegraph-test
+
+### Source and Intellectual Property Submission Plan
+
+The codes are currently under Apache License Version 2.0 and have been verified to there is no intellectual property or license issues before being released to open source by Baidu in 2018. Baidu will provide SGA and all committers will sign ICLA after HugeGraph is accepted into the Incubator.
+
+#### External Dependencies:
+
+As all dependencies are managed using Apache Maven, none of the external libraries need to be packaged in a source distribution. All dependencies have Apache compatible licenses except MySQL (GPL-2.0), we will remove MySQL dependencies in future.
```

Comment:
If the MySQL dependency is optional, we can add some instruction for users to download by themselves.

https://www.apache.org/legal/resolved.html#optional

Reply:
Yes, the MySQL dependency is optional.
TODO: add some instruction for users to download by themselves